### PR TITLE
Handle exceeding stock fulfillment approvals

### PR DESCRIPTION
**Pull request by Droniu**
I want to merge this change because it adds possibility to approve fulfillments which exceed their stock.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6455 Handle exceeding stock fulfillment approvals](https://app.clickup.com/t/2549495/SALEOR-6455)

File: src/orders/views/OrderDetails/OrderNormalDetails/index.tsx [hunk 1]
```diff
@@ -300,14 +324,39 @@ export const OrderNormalDetails: React.FC<OrderNormalDetailsProps> = ({
           []
         }
         open={params.action === "approve-fulfillment"}
-        onConfirm={({ notifyCustomer }) =>
-          orderFulfillmentApprove.mutate({
+        onConfirm={({ notifyCustomer }) => {
+          setCurrentApproval({
+            fulfillment: order?.fulfillments.find(
+              fulfillment => fulfillment.id === params.id
```
Comment (orzechdev) [inline src/orders/views/OrderDetails/OrderNormalDetails/index.tsx:330]: Might be util 
```suggestion
            fulfillment: order?.fulfillments.find(
              getById(params.id)
```

**Review by orzechdev — APPROVED** [review]

File: src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx [hunk 2]
```diff
@@ -35,15 +37,17 @@ const OrderFulfillStockExceededDialogLine: React.FC<OrderFulfillStockExceededDia
...
-        </Typography>
+        {"attributes" in line.variant && (
+          <Typography color="textSecondary" variant="caption">
+            {line.variant?.attributes
+              .map(attribute =>
+                attribute.values
+                  .map(attributeValue => attributeValue.name)
+                  .join(", ")
+              )
+              .join(" / ")}
```
Comment (jwm0) [inline src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx:48]: This should be a function

File: src/orders/components/OrderFulfillStockExceededDialog/OrderFulfillStockExceededDialog.tsx [hunk 3]
```diff
@@ -27,9 +24,9 @@ import { stockExceededDialogMessages as messages } from "./messages";
 import { useStyles } from "./styles";
 
 export interface OrderFulfillStockExceededDialogProps {
-  lines: OrderFulfillLineFragment[];
+  lines: Array<FulfillmentFragment["lines"][0] | OrderFulfillLineFragment>;
```
[the remark this answers is not in the document]
Comment (Droniu) [reply src/orders/components/OrderFulfillStockExceededDialog/OrderFulfillStockExceededDialog.tsx:27]: These types refer to different parts of code. `OrderFulfillLineFragment` is for lines in Order Fulfill view, which are from OrderFulfillDataQuery.

`FulfillmentFragment["lines"][0]` is data required for showing fulfillments in Order Details view

File: src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx (hunk 2 above)
Comment (Droniu) [reply src/orders/components/OrderFulfillStockExceededDialogLine/OrderFulfillStockExceededDialogLine.tsx]: Done

**Review by jwm0 — APPROVED** [review]

**Pull request merged**